packages/eslint-plugin/src/rules/no-use-before-define.ts

```diff
@@ -315,10 +315,13 @@ export default createRule<Options, MessageIds>({
       variable: TSESLint.Scope.Variable,
       reference: TSESLint.Scope.Reference,
     ): boolean {
-      return (
-        variable.identifiers[0].range[1] <= reference.identifier.range[1] &&
-        !isInInitializer(variable, reference)
-      );
+      if (variable.identifiers[0].range[1] <= reference.identifier.range[1]) {
+        if (reference.isValueReference) {
+          return !isInInitializer(variable, reference);
+        }
+        return true;
+      }
+      return false;
```

Comment:
[Style] Just "code golfing" for fun 😄 but if you're interested, one level of inlining..

```ts
      if (variable.identifiers[0].range[1] <= reference.identifier.range[1]) {
        return !(
          reference.isValueReference && isInInitializer(variable, reference)
        );
      }
      return false;
```

...or even two!

```ts
      return (
        variable.identifiers[0].range[1] <= reference.identifier.range[1] &&
        !(reference.isValueReference && isInInitializer(variable, reference))
      );
```

TBH I'm 50/50 on whether this is easier to read... but it is nice having just the one line 😄. @auvred also is 50/50.

Whatever makes you happiest. 😄

Reply:
I don't have a particular style preference for this code, I've fixed it with your second suggestion. 
35d0a18